AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+        self.keychain.synchronizable = false
+        self.userDefaults = userDefaults
+    }
+
+    // Async
+    func createAccount(completion: @escaping (Result<EthereumAccount, KeystoreError>) -> Void) {
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+            let result = strongSelf.createAccount()
+            DispatchQueue.main.async {
+                completion(result)
+            }
+        }
+    }
+
+    func importWallet(type: ImportType, completion: @escaping (Result<Wallet, KeystoreError>) -> Void) {
+        let results = importWallet(type: type)
+        switch results {
+        case .success(let wallet):
+            //TODO not the best way to do this but let's see if there's a better way to inform the coordinator that a wallet has been imported to avoid it being prompted for back
+            PromptBackupCoordinator(keystore: self, wallet: wallet, config: .init()).markWalletAsImported()
+        case .failure:
+            break
+        }
+        completion(results)
+    }
+
+    func importWallet(type: ImportType) -> Result<Wallet, KeystoreError> {
+        switch type {
+        case .keystore(let json, let password):
+            guard let keystore = try? LegacyFileBasedKeystore() else {
+                return .failure(.failedToExportPrivateKey)
+            }
+            let result = keystore.getPrivateKeyFromKeystoreFile(json: json, password: password)
+            switch result {
+            case .success(let privateKey):
+                return importWallet(type: .privateKey(privateKey: privateKey))
+            case .failure(let error):
+                return .failure(error)
+            }
+        case .privateKey(let privateKey):
+            let address = AlphaWallet.Address(fromPrivateKey: privateKey)
+            let hasEthereumAddressAlready = wallets.map({ $0.address }).contains {
+                $0.sameContract(as: address)
+            }
+            guard !hasEthereumAddressAlready else {
+                return .failure(.duplicateAccount)
+            }
+            if isUserPresenceCheckPossible {
+                let isSuccessful = savePrivateKeyForNonHdWallet(privateKey, forAccount: .init(address: address), withUserPresence: false)
+                guard isSuccessful else { return .failure(.failedToCreateWallet) }
+                let _ = savePrivateKeyForNonHdWallet(privateKey, forAccount: .init(address: address), withUserPresence: true)
+            } else {
+                let isSuccessful = savePrivateKeyForNonHdWallet(privateKey, forAccount: .init(address: address), withUserPresence: false)
+                guard isSuccessful else { return .failure(.failedToCreateWallet) }
+            }
+            addToListOfEthereumAddressesWithPrivateKeys(address)
+            return .success(Wallet(type: .real(.init(address: address))))
+        case .mnemonic(let mnemonic, _):
```

Comment:
Unlike in the Android version, the non-user authenticated ciphertext does not get deleted after the authenticated version gets stored. Is this intentional? Presumably, if the user imported the seed, he has access to it and we therefore don't need to worry about maintaining a non-user presence protected backup anymore. 

Is the reason that the importWallet method is also used when a new seed is created? In that case, I suggest using two separate methods, or using a flag to indicate that the non-user-presence backup should be removed if the user is importing a wallet from an existing seed (as opposed to creating a new one).

Reply:
Yes.

Upon importing a set of keys, the wallet key should still be accessible without user-presence (in case the user disables biometrics). It is only after the user choose to elevate security to make it go to "green bar" (refer to https://github.com/AlphaWallet/TokenScript/wiki/General-Nomenclature#green-bar-for-wallet-security-level for the odd, but unambiguous terminology)

Reply:
Thanks for clarifying